Add explicit steps for cases in rfc-0007-repository-ownership.md (#884)
cloudfoundry/community: toc/rfc/rfc-0007-repository-ownership.md

We illustrate several different cases of repository ownership changes and the approvals required for them to proceed:

#### Case 1: A Working Group wants to create a new repository under their ownership.

Approval requirements
* The Working Group MUST approve the addition of the proposed repository to the set of repositories that it owns.
* The TOC MUST approve the creation of the proposed repository.
* The TOC MUST approve the addition of the proposed repository to the Working Group's subset.

Steps

> 1. Someone opens two PRs. The frist PR adds the repo to the [cloudfoundry.yml](../../org/cloudfoundry.yml). The second PR adds the repo to the desired Working Group's yaml. These PRs should link to each other.

Author: ```suggestion
1. Someone opens two PRs. The first PR adds the repo to the [cloudfoundry.yml](../../org/cloudfoundry.yml). The second PR adds the repo to the desired Working Group's yaml. These PRs should link to each other.
```

> 1. Someone opens two PRs. The first PR adds the repo to the [cloudfoundry.yml](../../org/cloudfoundry.yml). The second PR adds the repo to the desired Working Group's yaml. These PRs should link to each other.

Reviewer: Maybe I'm missing something, but why do we need 2 PRs?

Author: I thought we needed to wait for the automation to create the repo before it was added to a WG. 

But if this isn't the case, I can change it.

Reviewer 2: One PR should be enough. Example: #821 - [automation run that created repo + teams](https://github.com/cloudfoundry/community/actions/runs/8927138863) (branchprotection got cancelled because of newer commits but peribolos run succeeded)

2. The Working Group and TOC both approve both PRs.
3. The TOC merges the first PR.
4. Wait for automation to run to create the repo.
5. The TOC merges the second PR.

#### Case 2: The TOC wants Working Group A to take ownership of an unclaimed active repository.

Approval requirements
* The TOC MUST approve the change in the ownership scope of Working Group A.
* Working Group A MUST approve the change to the set of repositories they own.

Steps
1. Someone opens a PRs that adds a repo to the desired Working Group's yaml.
2. The Working Group and TOC both approve the PR.
3. The TOC merges the PR.

#### Case 3: Working Group A wants to claim ownership of a repository currently owned by Working Group B.

Approval requirements

> * Both Working Group A and Working Group B MUST approve the change to the sets of repositories they own.

Reviewer: It would be great to call out which role and if it needs to be unanimous or if one vote per working group would be sufficient.

Author: This is defined above:

> Approval consists of a quorum decision by the leads of the Working Group.

* The TOC MUST also approve the transfer of the repository from Working Group B to Working Group A.

Steps
1. Someone opens a PR that moves a repo from Working Group A's yaml to Working Group B's yaml.
2. Working Group A, Working Group B, and TOC approve the PR.
3. The TOC merges the PR.

#### Case 4: A Working Group wants to archive a repository that it currently owns.

Approval requirements
* The Working Group MUST approve the removal of the repository from the set of repositories that it owns.
* The TOC MUST approve the removal of the repository from the Working Group and the archival.

Steps
1. Someone opens a PR that: marks the repo as archived in the [cloudfoundry.yml](../../org/cloudfoundry.yml) and that removes the repo from the Working Group's yaml.
2. The Working Group and TOC both approve the PR.
3. The TOC merges the PR.

#### Case 5: A Working Group wants to delete a repository that it currently owns.

> 

Reviewer 2: Maybe add a note: Deleting a repository will usually be rejected by the TOC. Please consider to archive the repository (Case 4).

> 

Author: ```suggestion
⚠️ Deleting a repository will usually be rejected by the TOC. Please consider archiving instead.

```

4. A github admin deletes the repo.

#### Case 6: A Working Group wants to cease maintaining a repository that it currently owns.

Approval requirements
* The Working Group MUST approve the removal of the repository from the set of repositories that it owns.
* The TOC MUST approve the removal of the repository from the Working Group.
* In addition, the TOC MUST decide whether to archive the repository or to arrange for another Working Group to own it instead (possibly a newly formed one, if appropriate and if supported by community contributors).
* Note: In this case, it is unlikely that either the TOC or the CFF staff should own this repository, as it was already owned by some Working Group in the CF community.

Steps
1. Someone opens a PR that removes the repo from the Working Group's yaml.
2. The TOC reviews and decides what action to take with the repo. Other PRs might be required for this step.
3. The Working Group and TOC both approve the PR.
5. The TOC merges the PR.

#### Case 7: A Working Group wants to rename a repository that it currently owns.

Approval requirements
* The Working Group A and the TOC MUST both approve of the new name for the repository.

Steps

> 1. Someone two PRs:

Reviewer 2: Someone **opens** two PRs

Author: ```suggestion
1. Someone opens two PRs:
```

   * PR 1: Add the new repo to the [cloudfoundry.yml](../../org/cloudfoundry.yml).
   * PR 2: Rename the repo in the Working Group's yaml. Remove the old repo from the [cloudfoundry.yml](../../org/cloudfoundry.yml).
The second PR adds the repo to the desired Working Group's yaml. These PRs should link to each other.
3. The Working Group and TOC both approve both PRs.
4. The TOC merges the first PR.
5. Wait for automation to run to create the repo.

> 6. A github admin copies the old repo contents to the new repo.

Reviewer 2: This renaming procedure seem risky. The org automation takes a long time so that there is easily a day between copying the repo content and publishing the new repo / deleting the old repo. Commits to the old repo after copying may get lost. Commits to the old repo are possible until the org automation has applied the second PR.

Maybe the old repo should be archived after copying immediately by the admin to reduce this risk. But it gets unarchived again by the next org automation run (before PR 2 is merged).

A possible alternative process:
- just one PR that renames the repo in cloudfoundy.yml and the WG yaml (note: this is something that the org automation can't handle correctly)
- WG and TOC approve the PR
- TOC stops org automation
- TOC merges PR
- TOC [renames the repo using the github UI](https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository) 
- TOC re-enables the org automation

Disclaimer: Haven't tested this procedure. If this manual procedure is not applied correctly, the org automation should fail (team assignment to non-existing repo).

Reviewer 3: This approach sounds more straight forward. Should a running org automation job be canceled or just preventing the next run is OK?

Reviewer 2: I would ensure that the org automation is not not running, neither with the an old commit nor with the merged rename PR. Otherwise the org automation may create unwanted empty repos that interfere with the rename. I guess that a running old workflow run would fail most of the time due to a missing repo - but with bad timing it may recreate the old repo.

A safe approach would be to disable the workflow and wait until a running org automation stops or, if its urgent, cancel a running automation.

Author: I like this approach. I will update the text.

We illustrate several different cases of repository ownership changes and the approvals required for them to proceed:

#### Case 1: A Working Group wants to create a new repository under their ownership.

Approval requirements
* The Working Group MUST approve the addition of the proposed repository to the set of repositories that it owns.
* The TOC MUST approve the creation of the proposed repository.
* The TOC MUST approve the addition of the proposed repository to the Working Group's subset.

Steps

> 1. Someone opens two PRs. The first PR adds the repo to the [cloudfoundry.yml](../../org/cloudfoundry.yml). The second PR adds the repo to the desired Working Group's yaml. These PRs should link to each other.
> 2. The Working Group and TOC both approve both PRs.
> 3. The TOC merges the first PR.
> 4. Wait for automation to run to create the repo.
> 5. The TOC merges the second PR.

Author: ```suggestion
1. Someone opens a PR that: adds the repo to the [cloudfoundry.yml](../../org/cloudfoundry.yml) and adds the repo to the desired Working Group's yaml.
2. The Working Group and TOC both approve the PR.
3. The TOC merges the first PR.
```

How's this @rkoster ?

4. A github admin deletes the repo.

#### Case 6: A Working Group wants to cease maintaining a repository that it currently owns.

Approval requirements
* The Working Group MUST approve the removal of the repository from the set of repositories that it owns.
* The TOC MUST approve the removal of the repository from the Working Group.
* In addition, the TOC MUST decide whether to archive the repository or to arrange for another Working Group to own it instead (possibly a newly formed one, if appropriate and if supported by community contributors).
* Note: In this case, it is unlikely that either the TOC or the CFF staff should own this repository, as it was already owned by some Working Group in the CF community.

Steps
1. Someone opens a PR that removes the repo from the Working Group's yaml.
2. The TOC reviews and decides what action to take with the repo. Other PRs might be required for this step.
3. The Working Group and TOC both approve the PR.
5. The TOC merges the PR.

#### Case 7: A Working Group wants to rename a repository that it currently owns.

Approval requirements
* The Working Group A and the TOC MUST both approve of the new name for the repository.

Steps

> 1. Someone opens two PRs:
>    * PR 1: Add the new repo to the [cloudfoundry.yml](../../org/cloudfoundry.yml).
>    * PR 2: Rename the repo in the Working Group's yaml. Remove the old repo from the [cloudfoundry.yml](../../org/cloudfoundry.yml).
> The second PR adds the repo to the desired Working Group's yaml. These PRs should link to each other.
> 3. The Working Group and TOC both approve both PRs.
> 4. The TOC merges the first PR.
> 5. Wait for automation to run to create the repo.
> 6. A github admin copies the old repo contents to the new repo.
> 7. The TOC merges the second PR.
> 8. A github admin deletes the old repo.

Author: @stephanme  - how is this?

```suggestion
1. Someone opens a PR that: renames the repo in the [cloudfoundry.yml](../../org/cloudfoundry.yml) and renames the repo in the Working Group's yaml.
2. The Working Group and TOC both approve the PR.
3. The TOC stops the Github automation.
4. The TOC merges the PR.
5. The TOC renames the repo using the Github UI.
6. The TOC re-enables the Github automation.
```
